src/createEpicMiddleware.ts

```diff
@@ -42,7 +41,7 @@ export function createEpicMiddleware<T extends Action, O extends T = T, S = void
     const stateSubject$ = new Subject<S>().pipe(
       observeOn(uniqueQueueScheduler)
     ) as any as Subject<S>;
```

Comment:
Nit: ` as any as Subject<S>` can be removed, now that a regular `Observable` is used instead of `ActionsObservable`.

I love this PR, as I think it simplifies both usage and the code quite a bit. Sorry if I just jumped in with this review.

Reply:
Wonderful! Thank you for pointing this out.

Reply:
Turns out it couldn't just be removed as-is because while observeOn() uses RxJS's internal lift() function so it does return a Subject, the type definition does not reflect that with pipe() so from the type system point of view it's just an Observable and doesn't have .next() nor .asObservable(). No biggie as I was able to refactor the code in another way to avoid the casts.